src/node_hybrid.cpp

```diff
@@ -365,13 +366,27 @@ bool NodeHybrid::isNodeValid(
   const bool & traverse_unknown,
   GridCollisionChecker * collision_checker)
 {
-  if (collision_checker->inCollision(
-      this->pose.x, this->pose.y, this->pose.theta /*bin number*/, traverse_unknown))
-  {
+  unsigned int map_x = static_cast<unsigned int>(this->pose.x + 0.5f);
+  unsigned int map_y = static_cast<unsigned int>(this->pose.y + 0.5f);
+  bool coordinates_in_bounds = (map_x < NodeHybrid::footprint_collision_map.info.width &&
+                                map_y < NodeHybrid::footprint_collision_map.info.height);
+
+  unsigned int index = map_y * NodeHybrid::footprint_collision_map.info.width + map_x;
+  bool index_valid = coordinates_in_bounds && (index < NodeHybrid::footprint_collision_map.data.size());
+
+  if (collision_checker->inCollision(this->pose.x, this->pose.y, this->pose.theta, traverse_unknown)) {
+    if (index_valid) {
+      NodeHybrid::footprint_collision_map.data[index] = 100; // Mark as occupied
+    }
```

Comment:
R: I think we should process this only if there is subscriber to the collision publisher
handle this in a cleaner way, i.e., do not add publisher checks all around